src/analysis/schwartz.lean

```diff
@@ -0,0 +1,400 @@
+/-
+Copyright (c) 2022 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+
+import analysis.calculus.cont_diff
+import analysis.normed_space.basic
+import analysis.locally_convex.with_seminorms
+import analysis.normed_space.multilinear
+import topology.algebra.uniform_filter_basis
+import analysis.inner_product_space.basic
+import tactic.positivity
+import algebra.order.pointwise
+
+/-!
+# Schwartz space
+
+This file defines the Schwartz space and the space of tempered distributions.
+
+## Main definitions
+
+* `schwartz`: The Schwartz space
+
+## Main statements
+
+* `schwartz.uniform_add_group` and `schwartz.locally_convex`: The Schwartz space is a locally
+convex topological vector space.
+
+## Notation
+
+* `𝓢(E, F)`: The Schwartz space `schwartz E F`
+
+## Tags
+
+Schwartz space, tempered distributions
+-/
+
+open filter
+open_locale big_operators ennreal nnreal topological_space
+
+noncomputable theory
+
+variables {R R' 𝕜 E F : Type*}
+
+variables [normed_add_comm_group E] [normed_space ℝ E]
+variables [normed_add_comm_group F] [normed_space ℝ F]
+
+variables (E F)
+
+/-- A function is a Schwartz function if it is smooth and all derivatives decay faster than
+  any power of ∥x∥. -/
+structure schwartz :=
```

Comment:
maybe `schwartz_map` (in the same way that a bundled continuous function is in `continuous_map`).